theano/tensor/nnet/abstract_conv.py

```diff
@@ -1517,18 +1525,23 @@ def conv(self, img, kern, mode="valid", dilation=1):
                      ] = kern
 
         if self.convdim == 2:
+            if img.shape[1] % self.num_groups != 0:
+                raise ValueError("Number of input channels must be divisible by num_groups")
             val = _valfrommode(mode)
             bval = _bvalfromboundary('fill')
+            input_channel_offset = img.shape[1] // self.num_groups
+            output_channel_offset = kern.shape[0]
```

Comment:
Thanks for checking! So TF actually only implements the special case of `num_groups == num_input_channels`. For this, they use a tensor of `[filter_height, filter_width, in_channels, channel_multiplier]`. Transposing to our layout, this is `[channel_multiplier, in_channels, filter_height, filter_width]`, which corresponds to `[out_channels // num_groups, in_channels, filter_height, filter_width]`. So yes, it's incompatible.

> So, shall I continue with the method being implemented in the new PR?

Seems like a good idea to me, yes!